event_model/__init__.py

```diff
@@ -180,6 +180,103 @@ def bulk_datum(self, doc):
         self.datum_page(bulk_datum_to_datum_page(doc))
 
 
+class SingleRunDocumentRouter(DocumentRouter):
+    """
+    A DocumentRouter intended to process events from exactly
+    one run.
+    """
+    def __init__(self):
+        self._start_doc = None
+        self._descriptors = dict()
```

Comment:
If we call `super().__init__()` here we can future-proof ourselves against possible changes to `DocumentRouter`.